src/resilience/openMP/OpenMPResParallel.hpp

```diff
@@ -177,5 +342,322 @@ class ParallelFor< FunctorType
 } // namespace Impl
 } // namespace Kokkos
 
+namespace Kokkos {
+namespace Impl {
+
+// MDRange policy implementation
+template <class FunctorType, class... Traits>
+class ParallelFor<FunctorType, 
+		  Kokkos::MDRangePolicy<Traits...>,
+                  KokkosResilience::ResOpenMP> {
+ private:
+  using MDRangePolicy = Kokkos::MDRangePolicy<Traits...>;
+  using Policy        = typename MDRangePolicy::impl_range_policy;
+  using WorkTag       = typename MDRangePolicy::work_tag;
+
+  using index_type   = typename Policy::index_type;
+  using iterate_type = typename Kokkos::Impl::HostIterateTile<
+      MDRangePolicy, FunctorType, typename MDRangePolicy::work_tag, void>;
+
+  const iterate_type m_iter;
+
+  const FunctorType &  m_functor;
+  const MDRangePolicy m_policy;
+
+  ParallelFor() = delete ;
+  ParallelFor & operator = ( const ParallelFor & ) = delete ;
+  
+  using surrogate_policy = Kokkos::MDRangePolicy < Kokkos::OpenMP, WorkTag>;
+
+ public:
+  inline void execute() const {
+    int repeats = 5;
+    bool success = 0;
+    while(success==0 && repeats > 0){
+  
+      surrogate_policy wrapper_policy;
+      wrapper_policy = surrogate_policy(m_policy.begin(), m_policy.end());
+  
+      // Trigger Subscriber constructors
+      KokkosResilience::ResilientDuplicatesSubscriber::in_resilient_parallel_loop = true;
+      auto m_functor_0 = m_functor;
+      auto m_functor_1 = m_functor;
+      KokkosResilience::ResilientDuplicatesSubscriber::in_resilient_parallel_loop = false;
+
+      Impl::ParallelFor< decltype(m_functor) , surrogate_policy, Kokkos::OpenMP > closure0(m_functor , wrapper_policy );
+      Impl::ParallelFor< decltype(m_functor) , surrogate_policy, Kokkos::OpenMP > closure1(m_functor_0 , wrapper_policy );
+      Impl::ParallelFor< decltype(m_functor) , surrogate_policy, Kokkos::OpenMP > closure2(m_functor_1 , wrapper_policy );
```

Comment:
I think we need to make MDRangePolicy have feature partiy with regular range policy; or if we want remove this from the PR and have in a different one